source/fundamentals/crud/read-operations/change-streams.txt

```diff
@@ -0,0 +1,266 @@
+
+Example
+~~~~~~~
+
+The following code example shows how to open a change stream and print
+change stream events whenever the data in the collection changes:
+
+.. literalinclude:: /includes/fundamentals/code-snippets/change-streams/ChangeStreams.java
+   :language: java
+   :dedent:
+   :start-after: begin openChangeStreamExample
+   :end-before: end openChangeStreamExample
+
+An insert operation on the collection should produce output similar to the
+following text:
+
+.. code-block::
+   :copyable: false
+
+   Received a change: ChangeStreamDocument{
+      operationType='insert',
+      resumeToken={"_data": "825EC..."},
+      namespace=myDb.myChangeStreamCollection,
+      ...
+   }
+
+For a runnable example, see the :ref:`<java-usage-watch>` usage example page.
+
+To learn more about the ``watch()`` method, see the following API
+documentation:
+
+- `MongoCollection.watch() <{+api+}/apidocs/mongodb-driver-sync/com/mongodb/client/MongoCollection.html#watch()>`__
+- `MongoDatabase.watch() <{+api+}/apidocs/mongodb-driver-sync/com/mongodb/client/MongoDatabase.html#watch()>`__
+- `MongoClient.watch() <{+api+}/apidocs/mongodb-driver-sync/com/mongodb/client/MongoClient.html#watch()>`__
+
+.. _java-change-stream-aggregation:
+
+Apply Aggregation Operators to your Change Stream
+-------------------------------------------------
+
+You can pass an aggregation pipeline as a parameter to the ``watch()`` method
+to specify which change events the change stream receives.
+
+To learn which aggregation operators your {+mdb-server+} version supports, see
+:ref:`<change-stream-modify-output>`.
+
+
+Example
+~~~~~~~
+
+The following code example shows how you can apply an aggregation pipeline to
+your change stream receive change events for insert and update operators
+only:
+
+.. literalinclude:: /includes/fundamentals/code-snippets/change-streams/ChangeStreams.java
+   :language: java
+   :dedent:
+   :start-after: begin aggregationExample
+   :end-before: end aggregationExample
+
+When the change stream receives an update change event, the preceding code
+example outputs the following text:
+
+.. code-block:: text
+   :copyable: false
+
+   Received a change to the collection: ChangeStreamDocument{
+   operationType=update, resumeToken={...},
+   ...
+
+.. _java-change-stream-configure-pre-post:
+
+Include Pre-images and Post-images
+----------------------------------
+
+You can configure the change event to contain or omit the following data:
+
+- The **pre-image** which is a document that represents the version of the
+  document before the operation if it exists
+- The **post-image** which is a document that represents the version of the
+  document after the operation if it exists
+
+To receive change stream events that include a pre-image or post-image, you
+must connect to a MongoDB v6.0 or later deployment and set up the following:
+
+- Enable pre-images and post-images for the collection on your MongoDB
+  deployment.
+
+  .. see::
```

Comment:
Suggestion: Since this admonition renders as a tip box with the title "See: " but is then followed by text rather than a link, it makes me think that there is a broken or missing link. Maybe it's just a matter of preference but I think this (and the following `.. see::` box) would be better as just a `.. tip::`